pyroute2/netlink/generic/wireguard.py

```diff
@@ -152,6 +158,10 @@ class wgdevice_peer(nla):
                        ('WGPEER_A_ALLOWEDIPS', 'wgpeer_a_allowedips'),
                        ('WGPEER_A_PROTOCOL_VERSION', 'uint32'))
 
+            def encode(self):
+                self['header']['type'] = 0x8000
```

Comment:
Can you try to set `nla_flags = NLA_F_NESTED` instead to overwrite encode here?

Reply:
You're right. Force pushed here d82533771a43ea5debca57adf5ab2e1a492098b1